pytorch_lightning/loggers/base.py

```diff
@@ -127,7 +127,7 @@ def agg_and_log_metrics(self, metrics: Dict[str, float], step: Optional[int] = N
         """
         agg_step, metrics_to_log = self._aggregate_metrics(metrics=metrics, step=step)
 
-        if metrics_to_log is not None:
+        if metrics_to_log is not None and metrics_to_log != {}:
```

Comment:
```suggestion
        if not metrics_to_log:
```
a possible simplification

Reply:
Thanks! But this wouldn't skip empty metric logs right? I tested it shortly with:

```python
for metrics_to_log in [None, {}]:

    # original        
    if metrics_to_log is not None and metrics_to_log != {}:
        print('[Simple] Don\'t print that for {}'.format(metrics_to_log))

    # simplified
    if metrics_to_log is not None:
        print('[Simple] Don\'t print that for {}'.format(metrics_to_log))
```

and got output:

```
[Simple] Don't print that for {}
```

In my opinion it makes sense to skip empty metric logs to avoid errors when logging empty dicts because of changes in the used libs in the loggers (e.g. wandb or tensorboard SummaryWriter). But this could be discussed :)

Reply:
ok. I was just thinking since you want to check if it's empty then there would be no need to check if it's a dict. like `len(metrics_to_log) != 0` for example would be more general

Reply:
Thanks for that suggestion. Isn't there a small bug?

I tested it shorlty with:
```python

for metrics_to_log in [None, {}]:

    # original        
    if metrics_to_log is not None and metrics_to_log != {}:
        print('[Original] Don\'t print that for {}'.format(metrics_to_log))

    # simplified
    if not metrics_to_log:
        print('[Simple] Don\'t print that for {}'.format(metrics_to_log))

    # correct
    if metrics_to_log:
        print('[Correct] Don\'t print that for {}'.format(metrics_to_log))
```

and got output:

```
[Simple] Don't print that for None
[Simple] Don't print that for {}
```

So correct would be?
```suggestion
        if metrics_to_log:
```

Reply:
Yes, that's what my brain was telling me but I wrote for some reason the ``not`` xD Sorry for the confusion.

Reply:
No problem :)